### PR TITLE
fix: keymap may not exist

### PR DESCRIPTION
**Pull request by tomtomjhj**

### CODE REVIEW

**Review by theHamsta — APPROVED** [review]

File: lua/nvim-treesitter/incremental_selection.lua [hunk 1]
```diff
@@ -157,9 +157,9 @@ function M.detach(bufnr)
   for f, mapping in pairs(config.keymaps) do
     if mapping then
       if f == "init_selection" then
-        vim.keymap.del("n", mapping, { buffer = bufnr })
+        pcall(vim.keymap.del, "n", mapping, { buffer = bufnr })
       else
-        vim.keymap.del("x", mapping, { buffer = bufnr })
+        pcall(vim.keymap.del, "x", mapping, { buffer = bufnr })
```
Comment (theHamsta) [inline lua/nvim-treesitter/incremental_selection.lua:162]: I'm bit worried that this hides an actual bug. Under which circumstance are you encountering this error?

**Review by theHamsta — CHANGES_REQUESTED** [review]

**Comment by tomtomjhj** [comment]
This is the bug I'm trying to fix.
`mini.vim`:
```vim
set rtp^=~/.vim/plugged/vim-fugitive
set rtp^=~/.vim/plugged/nvim-treesitter

lua << EOF
require'nvim-treesitter.configs'.setup {
  incremental_selection = {
    enable = true,
    keymaps = {
      init_selection = "<space><tab>",
      node_incremental = "<space><tab>",
      node_decremental = "<s-tab>",
      scope_incremental = "<space><s-tab>",
    },
  },
}
EOF
```
(assuming that lua parser is installed)
```
nvim --clean p-u mini.vim some-lua-file.lua
:GvdiffSplit
:q
:GvdiffSplit
:q
:GvdiffSplit
Error detected while processing function fugitive#Diffsplit[140]..function fugitive#Diffsplit[122]..BufReadCmd Autocommands for "fugitive://*"..BufReadPost Autocommands for "*":
Error executing lua callback: /usr/share/nvim/runtime/filetype.lua:21: Error executing lua: /usr/share/nvim/runtime/filetype.lua:22: function fugitive#Diffsplit[122]..BufReadCmd Autocommands for "fugitive://*"..BufReadPost Autocommands for "*"..FileType Autocommands for "*": Vim(append):Error executing lua callback: vim/keymap.lua:0: E31: No such mapping^@stack traceback:^@^I[C]: in function 'nvim_buf_del_keymap'^@^Ivim/keymap.lua: in function 'del'^@^I...treesitter/lua/nvim-treesitter/incremental_selection.lua:160: in function 'detach'^@^I.../plugged/nvim-treesitter/lua/nvim-treesitter/configs.lua:518: in function 'detach_module'^@^I.../plugged/nvim-treesitter/lua/nvim-treesitter/configs.lua:527: in function 'reattach_module'^@^I.../plugged/nvim-treesitter/lua/nvim-treesitter/configs.lua:131: in function <.../plugged/nvim-treesitter/lua/nvim-treesitter/configs.lua:130>^@^I[C]: in function 'nvim_cmd'^@^I/usr/share/nvim/runtime/filetype.lua:22: in function </usr/share/nvim/runtime/filetype.lua:21>^@^I[C]: in function 'nvim_buf_call'^@^I/usr/share/nvim/runtime/filetype.lua:21: in function </usr/share/nvim/runtime/filetype.lua:10>^@stack traceback:^@^I[C]: in function 'nvim_cmd'^@^I/usr/share/nvim/runtime/filetype.lua:22: in function </usr/share/nvim/runtime/filetype.lua:21>^@^I[C]: in function 'nvim_buf_call'^@^I/usr/share/nvim/runtime/filetype.lua:21: in function </usr/share/nvim/runtime/filetype.lua:10>^@stack traceback:^@^I[C]: in function 'nvim_buf_call'^@^I/usr/share/nvim/runtime/filetype.lua:21: in function </usr/share/nvim/runtime/filetype.lua:10>
```

I don't know why it happens, and I don't have enough bandwidth to investigate. 

In any case, I believe silencing the error from unmapping command is ok in general, because anyone at any time can remove some mapping. `b:undo_ftplugin` from many ftplugins does that too.

File: lua/nvim-treesitter/incremental_selection.lua (hunk 1 above)
Comment (cngu) [reply lua/nvim-treesitter/incremental_selection.lua:162]: @theHamsta Not the author, but I'm seeing this when using vim-unimpaired's `[l` and `]l` mappings to cycle through the location list.

**Comment by amaanq** [comment]
Just an FYI I do get this too when closing a buffer then going "back" to it sometimes

**Comment by tomtomjhj** [comment]
The root cause of this is fixed by #4415

**Pull request closed without merging**